### PR TITLE
Feature/3928 Add SMTP environment variables to the helm chart

### PR DESCRIPTION
**Pull request by sam-roberts-bjss**
Story: [AB#3928](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/3928)
Task: [AB#9450](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/9450)

### CODE REVIEW

File: charts/ordapi/templates/deployment.yaml [hunk 1]
```diff
@@ -60,6 +60,71 @@ spec:
                   key: {{ .Values.serviceDependencies.oidc.baseUriConfig.key | quote }}
             - name: ALLOWINVALIDCERTIFICATE
               value: {{ .Values.allowInvalidCertificate | quote }}
+            {{- with .Values.serviceDependencies.email }}
+            - name: SMTPSERVER__HOST
```
Comment (stuart-bass-bjss) [inline charts/ordapi/templates/deployment.yaml:64]: Please can you add these settings to appsettings.json, so it's clear that they can be set, with sensible defaults?

**Review by stuart-bass-bjss — CHANGES_REQUESTED** [review]

File: charts/ordapi/templates/deployment.yaml (hunk 1 above)
Comment (sam-roberts-bjss) [reply charts/ordapi/templates/deployment.yaml:64]: They are in a PR waiting for review & hopefully soon to be merged in, I just split the tasks out.

File: charts/ordapi/templates/deployment.yaml [hunk 2]
```diff
@@ -60,6 +60,71 @@ spec:
...
+              value: {{ .subjectPrefix | quote }}
+            {{- if .allowInvalidCertificate }}
+            - name: SMTPSERVER__ALLOWINVALIDCERTIFICATE
+              value: {{ .allowInvalidCertificate | quote }}
+            {{- end }}
+            - name: SMTPSERVER__AUTHENTICATION__ISREQUIRED
+              value: {{ .authenticationRequired | quote }}
+            {{- if .authenticationRequired }}
+            {{- with .userName }}
+            - name: SMTPSERVER__AUTHENTICATION__USERNAME
```
Comment (stuart-bass-bjss) [inline charts/ordapi/templates/deployment.yaml:81]: The username / pass aren't in appsettings.json

File: charts/ordapi/templates/deployment.yaml [hunk 3]
```diff
@@ -60,6 +60,71 @@ spec:
...
+            {{- if .recipientAddress }}
+            - name: PURCHASING__EMAILMESSAGE__RECIPIENT__ADDRESS
+              value: {{ .recipientAddress | quote }}
+            {{- end }}
+            {{- if .subject }}
+            - name: PURCHASING__EMAILMESSAGE__SUBJECT
+              value: {{ .subject | quote }}
+            {{- end }}
+            {{- if .htmlBody }}
+            - name: PURCHASING__EMAILMESSAGE__HTMLBODY
```
Comment (stuart-bass-bjss) [inline charts/ordapi/templates/deployment.yaml:119]: html body not in app settings

File: charts/ordapi/values.yaml [hunk 4]
```diff
@@ -81,8 +81,34 @@ serviceDependencies:
     baseUriConfig:
       name: 
       key: 
+  email:
+    hostUriConfig:
+      name:
+      key:  
+    port:
+    #allowInvalidCertificate: false # note, if not set, will default to the global allowInvalidCertificate setting for isapi
```
Comment (stuart-bass-bjss) [inline charts/ordapi/values.yaml:89]: this note mentions isapi, which is wrong.

**Review by stuart-bass-bjss — CHANGES_REQUESTED** [review]

**Review by stuart-bass-bjss — APPROVED** [review]

**Pull request merged**